statwrap/ux.py

```diff
@@ -40,20 +33,25 @@ class DataUploadWidget:
     >>>     uploader = DataUploadWidget('df3', auto_display=False)
     
     >>> display(output)
-    >>> display(uploader.uploader, uploader.submit_button)
+    >>> display(uploader)
     """
 
-    def __init__(self, variable_name, accept: str = '.csv,.xls,.xlsx,.xlsm,.xlsb,.odf,.ods,.odt', auto_display: bool = True):
-        self.accept = accept
+    def __init__(self, variable_name: str = 'df', auto_display: bool = True):
+        super().__init__()
+        self.accept = '.csv,.xls,.xlsx,.xlsm,.xlsb,.odf,.ods,.odt'
         self.supported = {'.csv', '.xls', '.xlsx', '.xlsm', '.xlsb', '.odf', '.ods', '.odt'}
-        self.uploader = widgets.FileUpload(accept=accept, multiple=False)
+        self.uploader = widgets.FileUpload(accept=self.accept, multiple=False)
         self.submit_button = widgets.Button(description=f"Create DF as \"{variable_name}\"")
         self.variable_name = variable_name
         self.auto_display = auto_display
 
-        self.uploadData()
+        self.submit_button.on_click(self._on_submit)
+        self.children = [self.uploader, self.submit_button]
+
+        if self.auto_display:
```

Comment:
Let's cut this and the `auto_display` param. That will prevent this behavior and we can address other usability improvements in a future PR.

![Screenshot 2024-06-23 at 11 15 00 AM](https://github.com/alexanderthclark/statwrap/assets/13120988/99f16b26-9866-47b7-a3c6-0e883219eb6f)